File: vega-time/get-time-with-Vega-API-client.py
#!/usr/bin/python3

"""
Script language: Python3

Talks to:
- Vega node (gRPC)

Apps/Libraries:
- gRPC: Vega-API-client (https://pypi.org/project/Vega-API-client/)
"""

# Note: this file uses smart-tags in comments to section parts of the code to
# show them as snippets in our documentation. They are not necessary to be
# included when creating your own custom code.
#
# Example of smart-tags:
#  __something:
# some code here
# :something__

import os

# __import_client:
import vegaapiclient as vac
# :import_client__

node_url_grpc = os.getenv("NODE_URL_GRPC")

# __get_time:
# Request the latest timestamp in nanoseconds since epoch from the Vega network
data_client = vac.VegaTradingDataClient(node_url_grpc)

# The "timestamp" field contains the resulting data we need.
vega_time = data_client.GetVegaTime(vac.api.trading.GetVegaTimeRequest())
print("Vega time:\n{}".format(vega_time))
# :get_time__

File: vega-time/get-time-with-Vega-API-client.go
package main

import (
	"fmt"
	"os"
	"time"

	api "code.vegaprotocol.io/protos/data-node/api/v1"
	"golang.org/x/net/context"
	"google.golang.org/grpc"
)

func main() {
	nodeURLGrpc := os.Getenv("NODE_URL_GRPC")
	if len(nodeURLGrpc) == 0 {
		panic("NODE_URL_GRPC is null or empty")
	}
	conn, err := grpc.Dial(nodeURLGrpc, grpc.WithInsecure())
	if err != nil {
		fmt.Println(err)
		return
	}
	defer conn.Close()

	// __get_time:
	// Request the latest timestamp in nanoseconds since epoch from the Vega network
	dataClient := api.NewTradingDataServiceClient(conn)
	request := api.GetVegaTimeRequest{}
	vegaTime, err := dataClient.GetVegaTime(context.Background(), &request)
	if err != nil {
		fmt.Println(err)
		return
	}

	// The "timestamp" field contains the resulting data we need.
	fmt.Printf("Vega time: %s", time.Unix(0, vegaTime.Timestamp))
	// :get_time__
}
